src/chainlit/server.py

```diff
@@ -301,14 +307,14 @@ def register_wildcard_route_handler():
     async def serve(path: str):
         """Serve the UI and public files."""
         if path:
-            public_file_path = os.path.join(config.root, "public", path)
-            ui_file_path = os.path.join(build_dir, path)
-
-            file_paths = [public_file_path, ui_file_path]
-
-            for file_path in file_paths:
-                if os.path.isfile(file_path):
-                    return FileResponse(file_path)
+            ui_file_path = Path(build_dir) / path
+
+            # Check if the file is within the intended directory
+            if (
+                ui_file_path.is_file()
+                and ui_file_path.resolve().parent in ui_file_path.resolve().parents
```

Comment:
I don't think this does anything since the parent will always be in all the parents?

Reply:
You are correct, I mean `and Path(build_dir).resolve() in ui_file_path.resolve().parents
`

Reply:
LGTM!